articles/building-a-cicd-pipeline-using-gitlab/index.md

```diff
@@ -0,0 +1,117 @@
+
+### Table of contents
+[Prerequisites](#prerequisites)
+[Setting up Gitlab](#setting-up-gitlab)
+[Building a CI/CD pipeline](#building-a-cicd-pipeline)
+[Code](#code)
+[Conclusion](#conclusion)
+
+### Prerequisites 
+1. Basic knowledge of Git and version control: https://www.section.io/engineering-education/beginner-guide-to-git/
+2. Basic understanding of CI/CD: https://www.section.io/engineering-education/what-is-jenkins/
+3. Basic knowledge of programming: https://www.python.org/about/gettingstarted/
+
+### Setting up Gitlab
+Setting up a Gitlab account is very easy. Head over to [this](https://gitlab.com/users/sign_in) link, create an account, or choose to sign in with Google or GitHub. Once you sign in, it will redirect you to the homepage where you can explore various options such as creating a new project, a group or just explore some public repositories. Follow these steps to set up a new project on Gitlab:
+
+1. On the homepage, click on "create blank project" to initialize a new repository. 
+![Create New](/engineering-education/building-a-cicd-pipeline-using-gitlab/create_new.png)
+
+2. Gitlab will redirect you to a form where you fill in the details of your project, such as title, description, etc. You can also choose to make your project public or private. After you finish adding the details, click on "create project". 
+![Blank Project](/engineering-education/building-a-cicd-pipeline-using-gitlab/blank_project.png)
+
+3. Click on "clone" and copy the HTTPS URL to clone the project. Open a terminal or command line and enter the following command to clone the project into your local machine:
+
+```bash
+git clone <https clone URL>
+```
+
+Now that you have the project on your local machine, we can start building our own CI/CD pipeline and deploy it on Gitlab using Gitlab-ci. 
+
+### Building a CI/CD pipeline
+[Continuous Integration](https://codeship.com/continuous-integration-essentials) can be defined as a process, in the software development lifecycle, where engineers and developers push code into a git-based repository. If developers do this very frequently, an automated process to verify and build the code is essential to save time and costs. This automated process of building and testing the code whenever a new commit is made is called CI or Continuous Integration. According to the [official documentation](https://docs.gitlab.com/ee/ci/), GitLab CI/CD is a powerful tool built into GitLab that allows you to apply all the continuous methods (Continuous Integration, Delivery, and Deployment) to your software with no third-party application or integration needed. 
```

Comment:
change "an automated process to verify and build the code is essential to save time and costs" to "an automated process, to verify and build the code, is included to save time and costs." 

remove the sentence "This automated process of building and testing the code whenever a new commit is made is called CI or Continuous Integration." it is redundant.